generation/core/generator.js

```diff
@@ -0,0 +1,205 @@
+      (carry, item) =>
+        item instanceof Array ? [...carry, ...item] : [...carry, item],
+      []
+    );
+    const typeChecks = allTypeChecks.filter(check => typeof check === "object");
+    const returnStatement = createReturnStatement(classJson, sanitizedJson);
+    return [...typeChecks, returnStatement];
+  }
+
+  function createTypeChecks(json) {
+    const checks = json.args.map(createTypeCheck);
+    checks.filter(check => Boolean(check));
+    return checks;
+  }
+
+  function addArgumentContentSanitizerCall(json) {
+    if (supportedContentSanitizersMap[json.type]) {
+      return supportedContentSanitizersMap[json.type].value(json.name);
+    }
+
+    return t.identifier(json.name);
+  }
+  function addArgumentTypeSanitizer(json) {
+    if (supportedContentSanitizersMap[json.type]) {
+      return supportedContentSanitizersMap[json.type].type;
+    }
+
+    return json.type;
+  }
+
+  // These types need no wrapping with {type: ..., value: }
+  const plainArgumentTypes = ["id<GREYMatcher>"];
+  function shouldBeWrapped({ type }) {
+    return !plainArgumentTypes.includes(type);
+  }
+  function createReturnStatement(classJson, json) {
+    const args = json.args.map(arg => shouldBeWrapped(arg) ?
+      t.objectExpression([
+        t.objectProperty(
+          t.identifier("type"),
+          t.stringLiteral(addArgumentTypeSanitizer(arg))
+        ),
+        t.objectProperty(
+          t.identifier("value"),
+          addArgumentContentSanitizerCall(arg)
+        )
+      ]) : addArgumentContentSanitizerCall(arg)
+    );
+
+    return t.returnStatement(
+      t.objectExpression([
+        t.objectProperty(
+          t.identifier("target"),
+          t.objectExpression([
+            t.objectProperty(t.identifier("type"), t.stringLiteral("Class")),
+            t.objectProperty(t.identifier("value"), t.stringLiteral(classValue(classJson)))
+          ])
+        ),
+        t.objectProperty(t.identifier("method"), t.stringLiteral(json.name)),
+        t.objectProperty(t.identifier("args"), t.arrayExpression(args))
+      ])
+    );
+  }
+
+  function createTypeCheck(json) {
+    const typeCheckCreator = typeCheckInterfaces[json.type];
+    const isListOfChecks = typeCheckCreator instanceof Array;
+    return isListOfChecks
+      ? typeCheckCreator.map(singleCheck => singleCheck(json))
+      : typeCheckCreator(json);
+  }
+
+  return function (files) {
+    Object.entries(files).forEach(([inputFile, outputFile]) => {
+      const input = fs.readFileSync(inputFile, "utf8");
+      const isObjectiveC = inputFile[inputFile.length - 1] === 'h';
+
+      const json = isObjectiveC ? objectiveCParser(input) : javaMethodParser(input);
+      const ast = t.program([createClass(json), createExport(json)]);
+      const output = generate(ast);
+
+      const commentBefore = "/**\n\n\tThis code is generated.\n\tFor more information see generation/README.md.\n*/\n\n";
```

Comment:
Maybe add a more visible comment, and warning, to make sure people won't try adding things manually.
something like 
```js
// This file is automatically generated.
// Do not modify this file -- YOUR CHANGES WILL BE ERASED!
```

Reply:
Yeah, sounds nice, will do this in a seperate small PR, to keep this change contained 👍 It's on my list now